Jenkinsfile

```diff
@@ -1,137 +1,32 @@
-  }
-}
-
-def deployPR(credentialsId, registry, imageName, tag, extraCommands) {
-  withKubeConfig([credentialsId: credentialsId]) {
-    def deploymentName = "$imageName-$tag"
-    sh "kubectl get namespaces $deploymentName || kubectl create namespace $deploymentName"
-    sh "helm upgrade $deploymentName --install --namespace $deploymentName --recreate-pods --atomic ./helm/$imageName --set image=$registry/$imageName:$tag $extraCommands"
-  }
-}
-
-def undeployPR(credentialsId, imageName, tag) {
-  withKubeConfig([credentialsId: credentialsId]) {
-    def deploymentName = "$imageName-$tag"
-    sh "helm delete --purge $deploymentName || echo error removing deployment $deploymentName"
-    sh "kubectl delete namespaces $deploymentName || echo error removing namespace $deploymentName"
-  }
-}
-
-def publishChart(registry, imageName, containerTag) {
-  // jenkins doesn't tidy up folder, remove old charts before running
-  sh "rm -rf helm-charts"
-  sshagent(credentials: ['helm-chart-creds']) {
-    sh "git clone git@gitlab.ffc.aws-int.defra.cloud:helm/helm-charts.git"
-    dir('helm-charts') {
-      sh 'helm init -c'
-      sh "sed -i -e 's/image: $imageName/image: $registry\\/$imageName:$containerTag/' ../helm/$imageName/values.yaml"
-      sh "helm package ../helm/$imageName"
-      sh 'helm repo index .'
-      sh 'git config --global user.email "buildserver@defra.gov.uk"'
-      sh 'git config --global user.name "buildserver"'
-      sh 'git checkout master'
-      sh 'git add -A'
-      sh "git commit -m 'update $imageName helm chart from build job'"
-      sh 'git push'
-    }
-  }
-}
 node {
   checkout scm
   try {
-    stage('Set branch, PR, and containerTag variables') {
-      (branch, pr, containerTag, mergedPrNo, repoUrl, commitSha) = getVariables(repoName)
-      if (pr ) {
-        sh "echo Building $pr"
-      } else if (branch == "master") {
-        sh "echo Building master branch"
-      } else {
-        currentBuild.result = 'ABORTED'
-        error('Build aborted - not a PR or a master branch')
-      }
-      updateGithubCommitStatus('Build started', 'PENDING', repoUrl, commitSha)
+    stage('Set PR, and containerTag variables') {
+      (pr, containerTag, mergedPrNo) = defraUtils.getVariables(repoName)
+      defraUtils.updateGithubCommitStatus('Build started', 'PENDING')
     }
     stage('Build test image') {
-      buildTestImage(imageName, BUILD_NUMBER)
+      defraUtils.buildTestImage(imageName, BUILD_NUMBER)
     }
     stage('Run tests') {
-      runTests(imageName, BUILD_NUMBER)
+      defraUtils.runTests(imageName, BUILD_NUMBER)
     }
-    // note: there should be a `build production image` step here,
-    // but the docker file is currently not set up to create a production only image
     stage('Push container image') {
-      pushContainerImage(registry, regCredsId, imageName, containerTag)
+      defraUtils.pushContainerImage(regCredsId, registry, imageName, containerTag)
```

Comment:
This should really be called "buildAndPushImage" instead, as it's not obvious that the build step is in here.